src/norm_flows.jl

```diff
@@ -0,0 +1,98 @@
+using Distributions
+using LinearAlgebra
+using Random
+using StatsFuns: softplus
+using Roots, LinearAlgebra # for inverse
+
+################################################################################
+#                            Planar and Radial Flows                           #
+#             Ref: Variational Inference with Normalizing Flows,               #
+#               D. Rezende, S. Mohamed(2015) arXiv:1505.05770                  #
+################################################################################
+
+mutable struct PlanarLayer <: Bijector
```

Comment:
Not sure how to accomplish your earlier comment and this at the same time. Maybe use `Union`?

Reply:
You can have composite types like below which doesn't explict have any constraints.

```julia
abstract type A end
struct B{T1,T2} <: A
    field1::T1
    filed2::T2
end
```

and this will preserve type information

```julia
typeof(B(1, [2])) # => B{Int64,Array{Int64,1}}
```